### PR TITLE
First content for config

### LINKED ISSUE #96: Provide first config content

### PR DESCRIPTION
**Pull request by ckittl**
Resolves #96

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -96,21 +98,21 @@ dependencies {
...
 
   // javafx
-  implementation group: 'org.openjfx', name: 'javafx-fxml', version: fxVersion
-  implementation group: 'org.openjfx', name: 'javafx-swing', version: fxVersion
-  implementation group: 'org.openjfx', name: 'javafx-graphics', version: fxVersion
-  implementation group: 'org.openjfx', name: 'javafx-web', version: fxVersion
+  implementation "org.openjfx:javafx-fxml:$fxVersion"
+  implementation "org.openjfx:javafx-swing:$fxVersion"
+  implementation "org.openjfx:javafx-graphics:$fxVersion"
+  implementation "org.openjfx:javafx-web:$fxVersion"
```
Comment (johanneshiry) [inline build.gradle:107]: Do we still needs this? I think its droppable as we do not want to have fx capabilities here anymore don't we? 
If this is only required for backward compatibility you may add these dependencies on the todo list in #76

File: src/main/resources/config/config-template.conf [hunk 2]
```diff
@@ -0,0 +1,26 @@
+input.osm: {
```
Comment (johanneshiry) [inline src/main/resources/config/config-template.conf:1]: I assume optionality due to different sources that will be available in the future?

File: src/main/resources/config/config-template.conf [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  #@optional
+  file: {
+    directory: String
+    hierarchic: Boolean | false
+  }
+}
+generation: {
+  #@optional
+  lv: {
+    distinctHouseConnections: Boolean | false
```
Comment (johanneshiry) [inline src/main/resources/config/config-template.conf:24]: what does this parameter do? Please add documentation behind it.

File: src/main/scala/edu/ie3/osmogrid/exception/IllegalConfigException.scala [hunk 4]
```diff
@@ -3,13 +3,10 @@
  * Institute of Energy Systems, Energy Efficiency and Energy Economics,
  * Research group Distribution grid planning and operation
  */
-package edu.ie3.osmogrid
 
-import spock.lang.Specification
+package edu.ie3.osmogrid.exception
 
-class DummySpec extends Specification {
```
Comment (johanneshiry) [inline src/main/scala/edu/ie3/osmogrid/exception/IllegalConfigException.scala:10]: 😂

File: src/test/resources/testConfig.conf [hunk 5]
```diff
@@ -0,0 +1,5 @@
+input.osm.pbf.file = "input_file_path"
+input.asset.file.directory = "asset_directory"
+input.asset.file.hierarchic = false
+output.file.directory = "output_file_path"
+generation.lv.distinctHouseConnections = true
```
Comment (johanneshiry) [inline src/test/resources/testConfig.conf:5]: Differs from default? By purpose?

File: src/test/scala/edu/ie3/osmogrid/cfg/OsmoGridConfigFactory.scala [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+ */
+
+package edu.ie3.osmogrid.cfg
+
+import com.typesafe.config.{Config, ConfigFactory}
+
+import java.io.File
+import scala.util.Try
+
+object OsmoGridConfigFactory {
```
Comment (johanneshiry) [inline src/test/scala/edu/ie3/osmogrid/cfg/OsmoGridConfigFactory.scala:14]: Why? 
I mean I see what it does but I don't see the reason why we need this...?

**Review by johanneshiry — CHANGES_REQUESTED** [review]
LGTM in general - only have a few minor questions

File: build.gradle (hunk 1 above)
Comment (ckittl) [reply build.gradle:107]: We don't need it. But I was bother by the kind of notation. ^^

File: src/main/resources/config/config-template.conf (hunk 2 above)
Comment (ckittl) [reply src/main/resources/config/config-template.conf:1]: Nah. You mandatorily need osm data, right? Everything within it should be optional (with boundary conditions, that only one of them is set).

File: src/main/resources/config/config-template.conf (hunk 3 above)
Comment (ckittl) [reply src/main/resources/config/config-template.conf:24]: Is added in rtd.. :eyes:

File: src/test/scala/edu/ie3/osmogrid/cfg/OsmoGridConfigFactory.scala (hunk 6 above)
Comment (ckittl) [reply src/test/scala/edu/ie3/osmogrid/cfg/OsmoGridConfigFactory.scala:14]: Just for testing. And to reduce code duplications in test classes.

File: src/test/resources/testConfig.conf (hunk 5 above)
Comment (ckittl) [reply src/test/resources/testConfig.conf:5]: Yes to check, that not the default value is used and the config is parsed correctly.

**Pull request merged**